tensorflow_serving/example/inception_client.cc

```diff
@@ -0,0 +1,179 @@
+#include <iostream>
+#include <fstream>
+
+#include <grpc++/create_channel.h>
+#include "tensorflow_serving/apis/prediction_service.grpc.pb.h"
+
+
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::ClientReader;
+using grpc::ClientReaderWriter;
+using grpc::ClientWriter;
+using grpc::Status;
+
+
+using tensorflow::serving::PredictRequest;
+using tensorflow::serving::PredictResponse;
+using tensorflow::serving::PredictionService;
+
+const std::string SERVING_HOST = "localhost:9000";
+
+const std::string INPUT_FILE_NAME = "/tmp/cropped_panda.jpg";
+const std::string SERVICE_NAME = "inception";
+
+
+typedef ::google::protobuf::Map< ::std::string, ::tensorflow::TensorProto > OutMap;
+
+
+class ServingClient {
+ public:
+  ServingClient(std::shared_ptr<Channel> channel)
+      : stub_(PredictionService::NewStub(channel)) {
+  }
+
+  std::string callPredict(std::string file_path){
+    PredictRequest predictRequest;
+    PredictResponse response;
+    ClientContext context;
+
+    predictRequest.mutable_model_spec()->set_name(SERVICE_NAME);
+
+    ::google::protobuf::Map< ::std::string, ::tensorflow::TensorProto >& inputs = 
+        *predictRequest.mutable_inputs();
+
+    ::tensorflow::TensorProto proto;
+
+    std::ifstream imageFile(file_path, std::ios::binary);
+
+    if (!imageFile.is_open()) {
+      std::cout << "Failed to open " << file_path << std::endl;
+      return "";
+    }
+
+    std::filebuf * pbuf = imageFile.rdbuf();
+    long fileSize = pbuf->pubseekoff(0, std::ios::end, std::ios::in);
+    
+    char* image = new char[fileSize];
+
+    pbuf->pubseekpos(0, std::ios::in);
+    pbuf->sgetn(image, fileSize);
+    imageFile.close();
+    
+
+    proto.set_dtype(::tensorflow::DataType::DT_STRING);
+    proto.add_string_val(image, fileSize);
+
+    std::cout << "image size:" << fileSize << std::endl;
+    proto.mutable_tensor_shape()->add_dim()->set_size(1);
+
+    inputs["images"] = proto;
+    
+    Status status = stub_->Predict(&context, predictRequest, &response);
+    if (status.ok()) {
+      std::cout << "call predict ok" << std::endl;
+      std::cout << "outputs size is "<< response.outputs_size() << std::endl;
+      OutMap& map_outputs =  *response.mutable_outputs();
+      OutMap::iterator iter;
+      int output_index = 0;
+      for(iter = map_outputs.begin();iter != map_outputs.end(); ++iter){
+        std::cout << "output " << output_index << " name is "<< iter->first << std::endl;
+
+        std::cout << "output " << output_index << " type is "<< 
```

Comment:
Looks like we can probably reuse helper methods from tensorflow/core/framework/tensor.cc, specifically 
Tensor::FromProto and Tensor::DebugString(). This should let you delete the code from here to approx line 120 of the current PR.